lib/cli/kit/config.rb

```diff
@@ -0,0 +1,148 @@
+require 'fileutils'
+
+module CLI
+  module Kit
+    class Config
+      XDG_CONFIG_HOME = 'XDG_CONFIG_HOME'
+
+      attr_reader :toolname
+      def initialize(toolname)
+        @toolname = toolname
+      end
+
+      # Returns the config corresponding to `name` from the config file
+      # `false` is returned if it doesn't exist
+      #
+      # #### Parameters
+      # `section` : the section of the config value you are looking for
+      # `name` : the name of the config value you are looking for
+      #
+      # #### Returns
+      # `value` : the value of the config variable (false if none)
+      #
+      # #### Example Usage
+      # `config.get('name.of.config')`
+      #
+      def get(section, name = nil)
+        section, name = section.split('.', 2) if name.nil?
+        # TODO: Remove this and all global configs
+        return get("global", section) if name.nil?
+        all_configs.dig("[#{section}]", name) || false
+      end
+
+      # Sets the config value in the config file
+      #
+      # #### Parameters
+      # `section` : the section of the config you are setting
+      # `name` : the name of the config you are setting
+      # `value` : the value of the config you are setting
+      #
+      # #### Example Usage
+      # `config.set('section', 'name.of.config', 'value')`
+      #
+      def set(section, name = nil, value)
+        section, name = section.split('.', 2) if name.nil?
+        # TODO: Remove this and all global configs
+        return set("global", section, value) if name.nil?
+        all_configs["[#{section}]"] ||= {}
+        all_configs["[#{section}]"][name] = value.nil? ? nil : value.to_s
+        write_config
+      end
+
+      def get_section(section)
+        (all_configs["[#{section}]"] || {}).dup
+      end
+
+      # Returns a path from config in expanded form
+      # e.g. shopify corresponds to ~/src/shopify, but is expanded to /Users/name/src/shopify
+      #
+      # #### Example Usage
+      # `config.get_path('srcpath', 'shopify')`
+      #
+      # #### Returns
+      # `path` : the expanded path to the corrsponding value
+      #
+      def get_path(section, name = nil)
+        v = get(section, name)
+        false == v ? v : File.expand_path(v)
+      end
+
+      def to_s
+        @ini.to_s
+      end
+
+      def un_ini
```

Comment:
This shouldn't be ported.  This was my escape hatch that I left in in case of problems.